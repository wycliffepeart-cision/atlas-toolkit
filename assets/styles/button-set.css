@layer atlas {
	atlas-button-set {
		--primary-spacing: 1em;
		--extended-gap: 1.25em;
		--space-around: 0.5em;
	}

	atlas-button-set {
		display: flex;
		flex-direction: column;

		margin-block: var(--space-around);
	}

	atlas-button-set::part(primary) {
		display: flex;
		justify-content: center;

		gap: var(--primary-spacing);
	}

	atlas-button-set::part(extended) {
		display: flex;
		justify-content: center;
	}

	atlas-button-set > [slot="extended"] {
		margin-block-start: var(--extended-gap);
	}
}
