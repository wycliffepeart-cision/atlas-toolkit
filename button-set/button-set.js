import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var ButtonSetTemplate = "<template>\n  <slot part=\"primary\"></slot>\n  <slot name=\"extended\" part=\"extended\"></slot>\n</template>\n";

let ButtonSet = class ButtonSet extends HTMLElement {
};
ButtonSet = __decorate([
    Template(ButtonSetTemplate)
], ButtonSet);

export { ButtonSet };
