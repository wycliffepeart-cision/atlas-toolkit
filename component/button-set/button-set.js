import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var ButtonSetTemplate = "<template>\n  <slot part=\"primary\"></slot>\n  <slot name=\"extended\" part=\"extended\"></slot>\n</template>\n";

let ButtonSet = class ButtonSet extends HTMLElement {
};
ButtonSet = __decorate([
    Template(ButtonSetTemplate)
], ButtonSet);
if (!customElements.get('atlas-button-set'))
    customElements.define('atlas-button-set', ButtonSet);

export { ButtonSet };
