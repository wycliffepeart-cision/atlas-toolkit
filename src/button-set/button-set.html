<template>
	<slot part="primary"></slot>
	<slot name="extended" part="extended"></slot>
</template>
