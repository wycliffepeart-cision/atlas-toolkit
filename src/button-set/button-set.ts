import ButtonSetTemplate from './button-set.html';
import { Template } from '../decorators/template';

@Template(ButtonSetTemplate)
export class ButtonSet extends HTMLElement {}
