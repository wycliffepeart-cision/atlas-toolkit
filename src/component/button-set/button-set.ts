import ButtonSetTemplate from './button-set.html';
import { Template } from '../../decorators/template';

@Template(ButtonSetTemplate)
export class ButtonSet extends HTMLElement {}

if (!customElements.get('atlas-button-set')) customElements.define('atlas-button-set', ButtonSet);