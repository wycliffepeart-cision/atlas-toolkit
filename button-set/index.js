import { ButtonSet } from './button-set.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-button-set'))
    customElements.define('atlas-button-set', ButtonSet);
