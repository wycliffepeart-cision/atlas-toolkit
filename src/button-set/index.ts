import { ButtonSet } from './button-set';

if (!customElements.get('atlas-button-set')) customElements.define('atlas-button-set', ButtonSet);
