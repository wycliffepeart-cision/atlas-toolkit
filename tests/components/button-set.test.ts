import '../../src/component/button-set/button-set';

const BUTTON_SET_TAG = 'atlas-button-set';

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();
});

describe('Button Set', () => {
  it('is correctly appended in the DOM', () => {
    const elem = document.createElement(BUTTON_SET_TAG);
    document.body.appendChild(elem);

    (expect(elem) as any).toBeInTheDocument();
  });

  it('is matches snapshot', () => {
    const elem = document.createElement(BUTTON_SET_TAG);
    document.body.appendChild(elem);

    (expect(elem.shadowRoot.innerHTML) as any).toMatchSnapshot();
  });
});
