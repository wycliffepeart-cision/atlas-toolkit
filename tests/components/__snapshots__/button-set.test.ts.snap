// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Set is matches snapshot 1`] = `
<slot part="primary">
</slot>
<slot name="extended"
      part="extended"
>
</slot>
`;
